src/FlaUILibrary/keywords/screenshot.py

```diff
@@ -2,34 +2,70 @@
 from robotlibcore import keyword
 from FlaUILibrary.flaui.module.screenshot import Screenshot
 from FlaUILibrary.robotframework import robotlog
-
+from FlaUILibrary.flaui.util.automationinterfacecontainer import AutomationInterfaceContainer
 
 class ScreenshotKeywords:
     """
     Interface implementation from Robotframework usage for screenshot keywords.
     """
 
-    def __init__(self, screenshots: Screenshot):
+    def __init__(self, screenshots: Screenshot, container: AutomationInterfaceContainer):
         """Creates screenshot keywords module to handle image capturing.
 
-        ``screenshots`` Screenshots module for image capturing.
+        ``screenshots`` Screenshots module for image capturing
+        ``container`` User automation container to handle element interaction
         """
         self._screenshots = screenshots
+        self._container = container
 
     @keyword
-    def take_screenshot(self):
-        """ Takes a screenshot of the whole desktop. Returns path to the screenshot if screenshot was created.
+    def set_screenshot_log_mode(self, log_mode: str):
+        """Sets the logging mode of the screenshot module. Default is 'File'.
+        Mode 'File' logs screenshots as files in the screenshot directory.
+        Mode 'Base64' logs screenshots as base64 encoded strings embedded in the test report.
+
+        Arguments:
+        | Argument      | Type   | Description          |
+        | log_mode      | string | File | Base64        |
+
+        Example:
+        | Set Screenshot Log Mode    Base64 |
+        """
+        self._screenshots.set_mode(log_mode)
+
+    @keyword
+    def take_screenshot(self, identifier=None, msg=None):
+        """ Takes a screenshot of the whole desktop. Returns screenshot depending on log mode.
+        Screenshot mode File -> returns filepath
+        Screenshot mode Base64 -> returns encoded base64 string of image
+
+        Arguments:
+        | Argument      | Type   | Description          |
+        | identifier    | string | XPath identifier from element |
+        | msg           | string | Custom error message          |
 
         Example:
         | Take Screenshot |
+        | Take Screenshot   <XPATH> |
+        | Take Screenshot   <XPATH>    "Your custom error message" |
         """
-        filepath = self._screenshots.execute_action(Screenshot.Action.CAPTURE,
+        image_var = None
+        if identifier:
+            module = self._container.create_or_get_module()
+            element = module.get_element(identifier, None, msg)
+            image_var = self._screenshots.execute_action(Screenshot.Action.CAPTURE_ELEMENT,

```

Comment:
This will always create a screenshot, even if the element could not be found. If the element is not found, the whole screen is capture like in default.
Might be better if the screenshot will not get created if the element was not found?